### PR TITLE
TPROD-321 - NotificationBundle - Web - Convert PHP Templates to Twig Templates

### PR DESCRIPTION
**Pull request by JoshuaEstes**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Configure One Signal Plugin with fake data
3. Under Channels look for Web notifications (it will be added after you enable the plugin)
4. Make sure to use all the actions there

- `/s/notifications` List
  - Does page load and render correctly with and without any notifications present
  - Sorting columns work correctly
  - Filtering work correctly
- `/s/notifications/new` New
  - Can new notifications be created
- `/s/notifications/view/{id}` Details
  - Page load/render correctly with no errors
- `/s/notifications/edit/{id}` Details
  - Can you edit the notification

### CODE REVIEW

File: app/bundles/NotificationBundle/Views/Notification/list_graph.html.twig [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{#
```
Comment (JoshuaEstes) [inline app/bundles/NotificationBundle/Views/Notification/list_graph.html.twig:1]: This file does not seem to be used anywhere

File: app/bundles/NotificationBundle/Views/Notification/template_graph.html.twig [hunk 2]
```diff
@@ -1,18 +1,8 @@
-<?php
```
Comment (JoshuaEstes) [inline app/bundles/NotificationBundle/Views/Notification/template_graph.html.twig:1]: This file does not seem to be used anywhere

**Review by mabumusa1 — COMMENTED** [review]
- [X] List of Notifications 
- [X] Create a New Notification
- [X] Create New Category 
- [X] Notification View
- [X] Notification Edit 
- [X] Search 

I think these features work as expected, when I tried to add campaign action to send web notifications, I got the following error, I am not sure if this is an old issue or a new one caused by this PR.

```[2023-02-03 08:08:03] mautic.CRITICAL: Uncaught PHP Exception ErrorException: "PHP Warning - Trying to access array offset on value of type null" at /var/www/html/app/bundles/NotificationBundle/Form/Type/NotificationSendType.php line 71 {"exception":"[object] (ErrorException(code: 0): PHP Warning - Trying to access array offset on value of type null at /var/www/html/app/bundles/NotificationBundle/Form/Type/NotificationSendType.php:71)
[stacktrace]
#0 /var/www/html/app/bundles/NotificationBundle/Form/Type/NotificationSendType.php(71): Mautic\\CoreBundle\\ErrorHandler\\ErrorHandler->handleError(2, 'PHP Warning - T...', '/var/www/html/a...', 71)
#1 /var/www/html/vendor/symfony/form/ResolvedFormType.php(131): Mautic\\NotificationBundle\\Form\\Type\\NotificationSendType->buildForm(Object(Symfony\\Component\\Form\\FormBuilder), Array)
#2 /var/www/html/vendor/symfony/form/Extension/DataCollector/Proxy/ResolvedTypeDataCollectorProxy.php(95): Symfony\\Component\\Form\\ResolvedFormType->buildForm(Object(Symfony\\Component\\Form\\FormBuilder), Array)
#3 /var/www/html/vendor/symfony/form/FormFactory.php(82): Symfony\\Component\\Form\\Extension\\DataCollector\\Proxy\\ResolvedTypeDataCollectorProxy->buildForm(Object(Symfony\\Component\\Form\\FormBuilder), Array)
#4 /var/www/html/vendor/symfony/form/FormBuilder.php(95): Symfony\\Component\\Form\\FormFactory->createNamedBuilder('properties', Object(Symfony\\Component\\Form\\Extension\\DataCollector\\Proxy\\ResolvedTypeDataCollectorProxy), NULL, Array)
#5 /var/www/html/vendor/symfony/form/FormBuilder.php(247): Symfony\\Component\\Form\\FormBuilder->create('properties', 'Mautic\\\\Notifica...', Array)
#6 /var/www/html/vendor/symfony/form/FormBuilder.php(197): Symfony\\Component\\Form\\FormBuilder->resolveChildren()
#7 /var/www/html/vendor/symfony/form/FormFactory.php(32): Symfony\\Component\\Form\\FormBuilder->getForm()
#8 /var/www/html/app/bundles/CampaignBundle/Controller/EventController.php(80): Symfony\\Component\\Form\\FormFactory->create('Mautic\\\\Campaign...', Array, Array)
#9 /var/www/html/app/bundles/CoreBundle/Controller/CommonController.php(458): Mautic\\CampaignBundle\\Controller\\EventController->newAction(0, '')
#10 /var/www/html/vendor/symfony/http-kernel/HttpKernel.php(159): Mautic\\CoreBundle\\Controller\\CommonController->executeAction('new', 0, 0, '')
#11 /var/www/html/vendor/symfony/http-kernel/HttpKernel.php(81): Symfony\\Component\\HttpKernel\\HttpKernel->handleRaw(Object(Symfony\\Component\\HttpFoundation\\Request), 1)
#12 /var/www/html/vendor/symfony/http-kernel/Kernel.php(201): Symfony\\Component\\HttpKernel\\HttpKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#13 /var/www/html/app/AppKernel.php(105): Symfony\\Component\\HttpKernel\\Kernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#14 /var/www/html/app/middlewares/CORSMiddleware.php(82): AppKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#15 /var/www/html/app/middlewares/HSTSMiddleware.php(35): Mautic\\Middleware\\CORSMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#16 /var/www/html/app/middlewares/CatchExceptionMiddleware.php(34): Mautic\\Middleware\\HSTSMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#17 /var/www/html/app/middlewares/Dev/IpRestrictMiddleware.php(52): Mautic\\Middleware\\CatchExceptionMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#18 /var/www/html/app/middlewares/VersionCheckMiddleware.php(58): Mautic\\Middleware\\Dev\\IpRestrictMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#19 /var/www/html/app/middlewares/TrustMiddleware.php(42): Mautic\\Middleware\\VersionCheckMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#20 /var/www/html/vendor/stack/builder/src/Stack/StackedHttpKernel.php(23): Mautic\\Middleware\\TrustMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#21 /var/www/html/vendor/stack/run/src/Stack/run.php(13): Stack\\StackedHttpKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request))
#22 /var/www/html/index_dev.php(25): Stack\
un(Object(Stack\\StackedHttpKernel))
#23 {main}
"} {"hostname":"mautic-web","pid":2860}```

**Review by mabumusa1 — APPROVED** [review]
I tested this PR again, it worked as expected

**Comment by TechBill** [comment]
PR review - Unable to create web notification - give a "Your request could not be processed." message when trying to save.

 
Screenshot 2023-02-08 at 2 07 55 PM

**Comment by mabumusa1** [comment]
Keeping the field `heading` empty will cause 500 error, we need better validation for NotifcaitonBundle. 

@TechBill was testing and left the `heading` field empty this caused this error 

```
Uncaught PHP Exception Doctrine\DBAL\Exception\NotNullConstraintViolationException: "An exception occurred while executing 'INSERT INTO push_notifications (is_published, date_added, created_by, created_by_user, date_modified, modified_by, modified_by_user, checked_out, checked_out_by, checked_out_by_user, name, description, lang, url, heading, message, button, utm_tags, notification_type, publish_up, publish_down, read_count, sent_count, mobile, mobileSettings, category_id) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)' with params [1, "2023-02-08 20:20:20", 1, "Admin Mautic", null, null, null, null, null, null, "test one", null, "en", null, null, "one and two", null, "a:4:{s:9:\"utmSource\";N;s:9:\"utmMedium\";N;s:11:\"utmCampaign\";N;s:10:\"utmContent\";N;}", "template", null, null, 0, 0, 0, "N;", null]: SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'heading' cannot be null" at /var/www/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 125
```

**Comment by TechBill** [comment]
Tested PR - Cannot leave empty field under "Heading" otherwise will not save notification even if it's not marked as required.  Suggest either "Heading" required or fix that Heading field can be left empty when saved. 


```
/s/notifications List
Does page load and render correctly with and without any notifications present
Sorting columns work correctly
Filtering work correctly
```
**PASSED**


```
/s/notifications/new New
Can new notifications be created
```
**PASSED**

```
/s/notifications/view/{id} Details
Page load/render correctly with no errors
```
**PASSED**


```
/s/notifications/edit/{id} Details
Can you edit the notification
```
**PASSED**

**Pull request merged**